internal/orchestrator/orchestrator.go

```diff
@@ -295,7 +314,38 @@ func (or *orchestrator) initPlugins(ctx context.Context) (err error) {
 			return err
 		}
 	}
-	return or.dataexchange.Init(ctx, dataexchangeConfig.SubPrefix(or.dataexchange.Name()), &or.bc)
+	if err = or.dataexchange.Init(ctx, dataexchangeConfig.SubPrefix(or.dataexchange.Name()), &or.bc); err != nil {
+		return err
+	}
+
+	if or.tokens == nil {
+		or.tokens = make(map[string]tokens.Plugin)
+		for i := 0; i < tokensConfig.ArraySize(); i++ {
+			prefix := tokensConfig.ArrayEntry(i)
+			name := prefix.GetString("name")
+			connector := prefix.GetString("connector")
+			if name == "" || connector == "" {
+				return i18n.NewError(ctx, i18n.MsgMissingTokensPluginConfig)
+			}
+
+			log.L(ctx).Infof("Loading tokens plugin name=%s connector=%s", name, connector)
+			plugin, err := tkfactory.GetPlugin(ctx, connector)
+			if err != nil {
+				return err
+			}
+			or.tokens[name] = plugin
+		}
+	}
+	i := 0
+	for _, plugin := range or.tokens {
```

Comment:
Any reason not to just init these in the loop above?

Reply:
Seemed like all the previous logic was designed so that this method could be called with any plugin already set to non-nil, and it would just invoke Init without doing the discovery of plugins. At least that's how the unit tests are structured - so I added separate tests for discovery and for init of the plugin.